// INTEL CONFIDENTIAL
//
// Copyright (C) 2023 Intel Corporation
//
// This software and the related documents are Intel copyrighted materials, and your use of them is governed by
// the express license under which they were provided to you ("License"). Unless the License provides otherwise,
// you may not use, modify, copy, publish, distribute, disclose or transmit this software or the related documents
// without Intel's prior written permission.
//
// This software and the related documents are provided as is, with no express or implied warranties,
// other than those that are expressly stated in the License.

syntax = "proto3";

option go_package = "geti.com/account_service_grpc/pb";

package organization_status;

import "google/protobuf/timestamp.proto";
import "organization.proto";

service OrganizationStatus {
  // PUT /organizations/{id}/statuses
  rpc change (OrganizationStatusRequest) returns (OrganizationStatusResponse) {}

  // GET /organizations/{id}/statuses
  rpc get_statuses (organization.OrganizationIdRequest) returns (ListOrganizationStatusesResponse) {}
}

message OrganizationStatusRequest {
  string status = 1;
  string organization_id = 2;
  string created_by = 3;
}

message OrganizationStatusResponse {
  string id = 1;
  string status = 2;
  string organization_id = 3;
  google.protobuf.Timestamp created_at = 4;
  string created_by = 5;
}

message ListOrganizationStatusesResponse {
  repeated OrganizationStatusResponse statuses = 1;
}
